recipe-client-addon/lib/EventEmitter.jsm

```diff
@@ -20,15 +19,13 @@ this.EventEmitter = function(driver) {
       Promise.resolve()
         .then(() => {
           if (!(eventName in listeners)) {
-            driver.log(`EventEmitter: Event fired with no listeners: ${eventName}`);
+            log.info(`EventEmitter: Event fired with no listeners: ${eventName}`);
             return;
           }
-          // freeze event to prevent handlers from modifying it
-          const frozenEvent = Object.freeze(event);
```

Comment:
It does! And that _almost_ works.

The problem with cloning the object into the sandbox is that we can't call `emit` on the cloned emitter without calling `waiveXrays` on it, and if we do that, then the resulting emitter in the privileged compartment can have callbacks attached to it, which the sandbox could then trigger by calling `emit` on the cloned emitter, which seems like a bad idea for potential future engineers who think adding callbacks to the emitter is an easy fix for some problem.

So, I tried re-introducing the sandbox parameter to the emitter, and creating a new emitter specifically for the sandbox that doesn't have `emit`. I'm not entirely sure how to reason about potential leaks now that we've reintroduced the sandbox parameter to the event emitter, so I wanted to check with you. 

@gijsk How does this latest commit look?